generator/src/main/java/com/algorand/sdkutils/generators/Generator.java

```diff
@@ -215,57 +205,65 @@ static TypeDef getType(
         if (!format.isEmpty() ) {
             switch (format) {
             case "uint64":
-                return new TypeDef("java.math.BigInteger");
+                return new TypeDef("java.math.BigInteger", type, propName, goName, desc, isRequired(prop));
             case "RFC3339 String":
                 addImport(imports, "java.util.Date");
                 addImport(imports, "com.algorand.algosdk.v2.client.common.Utils");
-                return new TypeDef("Date");
+                return new TypeDef("Date", "time", propName, goName, desc, isRequired(prop));
             case "Address":
-                return getAddress(propName, imports, forModel);
+                return getAddress(propName, goName, imports, forModel, desc, isRequired(prop));
             case "SignedTransaction":
                 addImport(imports, "com.algorand.algosdk.transaction.SignedTransaction");
-                return new TypeDef("SignedTransaction");
+                return new TypeDef("SignedTransaction", type, propName, goName, desc, isRequired(prop));
             case "binary":
-                return getBase64Encoded(propName, null, forModel);
+                return getBase64Encoded(propName, goName, type, null, forModel, desc, isRequired(prop));
             case "byte":
             case "base64":
             case "digest":
                 if (type.contentEquals("array")) {
-                    return getBase64EncodedArray(propName, imports, forModel);
+                    type = prop.get("items").get("type").asText(); 
+                    return getBase64EncodedArray(propName, goName, type, imports, forModel, desc, isRequired(prop));
                 } else {
-                    return getBase64Encoded(propName, imports, forModel);
+                    return getBase64Encoded(propName, goName, type, imports, forModel, desc, isRequired(prop));
                 }
             case "AccountID":
                 break;
             case "BlockCertificate":
             case "BlockHeader":
                 addImport(imports, "java.util.HashMap");
-                return new TypeDef("HashMap<String,Object>");
+                return new TypeDef("HashMap<String,Object>", type, propName, goName, desc, isRequired(prop));
             }
         }
         switch (type) {
         case "integer":
-            return asObject ? new TypeDef("Long") : new TypeDef("long");
+            String longName = asObject ? "Long" : "long"; 
+            return new TypeDef(longName, type, propName, goName, desc, isRequired(prop));
         case "object":
```

Comment:
I'm not sure it's valid to map object -> string, maybe we don't actually hit this case currently?

Reply:
e.g. HealthCheck hits this. This will be refined in due time (i.e. testing for applications).